src/visibility.js

```diff
@@ -0,0 +1,165 @@
+    }
+
+    this._addBindings();
+    const options = {
+      threshold: this.config.threshold
+    };
+    this._observer = new window.IntersectionObserver(this._handleVisibilityChange.bind(this), options);
+    this._observer.observe(this._appTargetContainer);
+  }
+
+  _initFloating() {
+    this.config = Utils.Object.mergeDeep(DEFUALT_FLOATING_CONFIG, Utils.Object.copyDeep(this.config));
+    this._floatingPoster = Utils.Dom.createElement('div');
+    this._floatingPoster.className = FLOATING_POSTER_CLASS;
+    this._floatingContainer = Utils.Dom.createElement('div');
+    this._floatingContainer.className = FLOATING_CONTAINER_CLASS;
+
+    this._appTargetContainer = Utils.Dom.getElementById(this.player.config.targetId);
+    Utils.Dom.prependTo(this._floatingPoster, this._appTargetContainer);
+    let kalturaPlayerContainer = Utils.Dom.getElementById(this.player.config.ui.targetId);
+    this._appTargetContainer.replaceChild(this._floatingContainer, kalturaPlayerContainer);
+    Utils.Dom.appendChild(this._floatingContainer, kalturaPlayerContainer);
+
+    this.config.floating.position.split('-').forEach(side => {
+      Utils.Dom.addClassName(this._floatingContainer, `${FLOATING_ACTIVE_CLASS}-${side}`);
+    });
+  }
+
+  _dismissed() {
+    this._dismissed = true;
+    this.player.pause();
+    this._stopFloating();
+  }
+
+  _stopFloating() {
+    Utils.Dom.removeClassName(this._floatingContainer, FLOATING_ACTIVE_CLASS);
+    Utils.Dom.removeAttribute(this._floatingContainer, 'style');
+  }
+
+  _startFloating() {
+    Utils.Dom.addClassName(this._floatingContainer, FLOATING_ACTIVE_CLASS);
+    Utils.Dom.setStyle(this._floatingContainer, 'height', this.config.floating.size.height);
+    Utils.Dom.setStyle(this._floatingContainer, 'width', this.config.floating.size.width);
+    Utils.Dom.setStyle(this._floatingContainer, 'margin', `${this.config.floating.margin.y} ${this.config.floating.margin.x}`);
+  }
+
+  _handleVisibilityChange(entries: Array<window.IntersectionObserverEntry>) {
+    const playerIsOutOfVisibility = entries[0].intersectionRatio < this.config.threshold;
+    if (this.config.floating && this._everStartedPlaying && !this._dismissed) {
+      this._handleFloatingChange(playerIsOutOfVisibility);
+    }
+  }
+
+  _handleFloatingChange(playerIsOutOfVisibility: boolean) {
+    if (playerIsOutOfVisibility) {
+      this._startFloating();
+    } else {
+      this._stopFloating();
+    }
+  }
+  /**
+   * _addBindings
+   * @private
+   * @returns {void}
+   */
+  _addBindings(): void {
+    this.eventManager.listen(this.player, this.player.Event.FIRST_PLAYING, () => {
+      this._everStartedPlaying = true;
+      Utils.Dom.setStyle(this._floatingPoster, 'background-image', `url("${this.player.config.sources.poster}")`);
+    });
+  }
+
+  /**
+   * Destroys the plugin.
+   * @override
+   * @public
+   * @returns {void}
+   */
+  destroy(): void {
+    this.logger.debug('destroy');
+    this._observer.disconnect();
```

Comment:
clear the `this._observer`, clear could be by setting the value to null

Reply:
I think you should revert the DOM change, maybe just remove the extra nodes is good enough.
And clear the DOM reference as well